Simplifying the device tree binding s for NXP's SoC's

yvanderv opened the pull request:
This series of commits reduces number of the device tree bindings to reflect the actual hardware.

danieldegrasse commented on dts/bindings/flash_controller/nxp,iap-f1.yaml:
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2019, NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP (In-Application Programming) flash memory controller for lpc54114
```
Can you reduce the length of this commit message title, so that it does not overflow into the commit body?

dleach02 commented on dts/bindings/flash_controller/nxp,iap-f2.yaml:
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2019, NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP (In-Application Programming) flash memory controller for all kinetis and most lpc devices
```
seems pedantic but should we consider explicitly listing the "most lpc devices"?

yvanderv replied:
OK, makes sense

danieldegrasse commented on drivers/flash/Kconfig.mcux:
```diff
@@ -3,7 +3,7 @@
 	bool "MCUX flash shim driver"
 	default y
-	depends on DT_HAS_NXP_IAP_F2_ENABLED 
+	depends on DT_HAS_NXP_IAP_F2_ENABLED
```
This change should be squashed into prior commit

galak commented on dts/bindings/flash_controller/nxp,iap-f1.yaml:
```diff
@@ -0,0 +1,8 @@
+
+description: NXP (In-Application Programming) flash memory controller for lpc54114
+
+compatible: "nxp,iap-f1"
```
What does `f1` mean?

galak commented on dts/bindings/flash_controller/nxp,iap-f2.yaml:
```diff
@@ -0,0 +1,8 @@
+
+description: NXP (In-Application Programming) flash memory controller for all kinetis and most lpc devices
+
+compatible: "nxp,iap-f2"
```
what does f2 mean?

galak reviewed (changes requested):
I'm not sure this makes sense and is possibly going the wrong way.  The compatibles should represent the HW programming model and not a HAL.

yvanderv committed:
dts: bindings: introducing new bindings for NXP's on-chip flash controllers

The binding 'nxp,iap-lpc' is used for different flash controllers,
preventing clarity. This commit introduces bindings for each version of
the flash controller, as they appear with NXP lpc device families:

nxp,iap-fmc11: for the flash controler used on the lpc11u6x family
nxp,iap-fmc54: for the flash controler used on the lpc54xx family
nxp,iap-fmc55: for the flash controler used on the lpc55xx family,
   except lpc553x
nxp,iap-fmc553: for the flash controler used on the lpc553x family

Signed-off-by: Yves Vandervennet <yves.vandervennet@nxp.com>

yvanderv committed:
flash: nxp: updating drivers, Kconfig and device trees with new bindings

The lpc and mcux drivers' DRV_COMPAT is updated for the new bindings
introduced in the previous commit. The drivers' Kconfig files also
reflect this change (DT_HAS_ENABLED_NXP_...).
The SoC device trees are updated with the new bindings

Signed-off-by: Yves Vandervennet <yves.vandervennet@nxp.com>

yvanderv committed:
flash: nxp: removal of deprecated binding nxp,lpc-iap

The binding 'nxp,lpc-iap' is no longer used, which is confirmed
by running:

$ find ${ZEPHYR_BASE}/dts/arm/nxp -type f | egrep -e '\.dts(i)*$' | \
  xargs grep -nH nxp,lpc-iap

Changes in this commit:
 - remove DT_HAS_NXP_LPC... in drivers/flash/Kconfig.mcux
 - remove schema file for nxp,lpc-iap

Signed-off-by: Yves Vandervennet <yves.vandervennet@nxp.com>
